sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -32,9 +106,30 @@ case class GpuGetJsonObject(json: Expression, path: Expression)
   override def nullable: Boolean = true
   override def prettyName: String = "get_json_object"
 
-  override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {    
-    lhs.getBase().getJSONObject(rhs.getBase, 
-    GetJsonObjectOptions.builder().allowSingleQuotes(true).build());
+  private var cachedNormalizedPath: Option[Option[String]] = None
+
+  def normalizeJsonPath(path: GpuScalar): Option[String] = {
+    val pathStr = if (path.isValid) {
+      path.getValue.toString()
+    } else {
+      throw new IllegalArgumentException("Invalid path")
```

Comment:
I don't know how big of a deal this is, but it means that the path is null. We should not throw an exception here, but we should return `None` as a null path should result in a null for all resulting values instead of an exception.

Reply:
done